### PR TITLE
main is now v3. What used to be main is now in the v2 branch.

### PR DESCRIPTION
**Pull request by BertKleewein**

### CODE REVIEW

File: azure-iot-device/azure/iot/device/constant.py [hunk 1]
```diff
@@ -6,17 +6,17 @@
 """This module defines constants for use across the azure-iot-device package
 """
 
-VERSION = "2.12.0"
+VERSION = "3.0.0b2"
 IOTHUB_IDENTIFIER = "azure-iot-device-iothub-py"
 PROVISIONING_IDENTIFIER = "azure-iot-device-provisioning-py"
-IOTHUB_API_VERSION = "2019-10-01"
+IOTHUB_API_VERSION = "2020-09-30"
 PROVISIONING_API_VERSION = "2019-03-31"
```
Comment (olivakar) [inline azure-iot-device/azure/iot/device/constant.py:13]: Is there a new one for this ?

File: migration_guide_provisioning.md [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+This guide details how to update existing code for IoT Hub provisioning that uses an `azure-iot-device` V2 release to use a V3 release instead.
+
+**Note that currently V3 only presents an async set of APIs. This guide will be updated when that changes**
+
+**Note that ProvisioningSession is still not yet available as it is in development**
+
+For changes when communicating between a device and IoT Hub, please refer to `migration_guide_iothub.md` in this same directory.
+
+## Provisioning using Shared Access Key (Symmetric Key)
+Using shared access key authentication (formerly called symmetric key) is now provided via the `shared_access_key` parameter on the `ProvisioningSession` constructor. rather than having it's own `create_from_symmetric_key()` factory method.
```
Comment (olivakar) [inline migration_guide_provisioning.md:12]: This information is wrong based on the last discussion we had. I had the class method create from symmetric key but now I don't have it anymore.

File: tests/e2e/provisioning_e2e/tests/test_async_certificate_enrollments.py [hunk 3]
```diff
@@ -300,4 +299,6 @@ async def result_from_register(registration_id, device_cert_file, device_key_fil
         websockets=protocol_boolean_mapping[protocol],
     )
 
-    return await provisioning_device_client.register()
+    result = await provisioning_device_client.register()
```
Comment (olivakar) [inline tests/e2e/provisioning_e2e/tests/test_async_certificate_enrollments.py:302]: were these files changed ? i thought the gate was shut off from running

**Review by olivakar — APPROVED** [review]

**Comment by robrennie** [comment]
Where is the new thermostat example? There do not appear to be any samples related to digital twins? Is there a migration guide for this?

**Pull request merged**